tests/integration/alibaba_oss/test_alibaba_oss.py

```diff
@@ -0,0 +1,212 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import random
+import tempfile
+import uuid
+from pathlib import Path
+
+import pandas as pd
+import pytest
+import yaml
+from pandas.testing import assert_frame_equal
+
+import mlrun
+import mlrun.errors
+
+from mlrun.secrets import SecretsStore
+from mlrun.utils import logger
+
+here = Path(__file__).absolute().parent
+config_file_path = here / "test-alibaba-oss.yml"
+with config_file_path.open() as fp:
+    config = yaml.safe_load(fp)
+
+test_filename = here / "test.txt"
+with open(test_filename) as f:
+    test_string = f.read()
+
+# Used to test dataframe functionality (will be saved as csv)
+test_df_string = "col1,col2,col3\n1,2,3"
+
+credential_params = ["ALIBABA_ACCESS_KEY_ID", "ALIBABA_SECRET_ACCESS_KEY","ALIBABA_ENDPOINT_URL"]
+
+
+def alibaba_oss_configured(extra_params=None):
+    extra_params = extra_params or []
+    env_params = config["env"]
+    needed_params = ["bucket_name", *credential_params, *extra_params]
+    for param in needed_params:
+        if not env_params.get(param):
+            return False
+    return True
+
+
+@pytest.mark.skipif(not alibaba_oss_configured(), reason="ALIBABA OSS parameters not configured")
+@pytest.mark.parametrize("use_datastore_profile", [False])
+class TestALIBABAOSS:
+    def _make_target_names(
+        self, prefix, bucket_name, object_dir, object_file, csv_file
+    ):
+        bucket_path = prefix + bucket_name
+        object_path = f"{object_dir}/{object_file}"
+        df_path = f"{object_dir}/{csv_file}"
+        object_url = f"{bucket_path}/{object_path}"
+        res = {
+            "bucket_path": bucket_path,
+            "object_path": object_path,
+            "df_path": df_path,
+            "object_url": object_url,
+            "df_url": f"{bucket_path}/{df_path}",
+            "blob_url": f"{object_url}.blob",
+            "parquet_url": f"{object_url}.parquet",
+            "bucket_path_dir":f"{bucket_path}/{object_dir}"
+        }
+        return res
+
+    def setup_method(self, method):
+        self._bucket_name = config["env"].get("bucket_name")
+        self._access_key_id = config["env"].get("ALIBABA_ACCESS_KEY_ID")
+        self._secret_access_key = config["env"].get("ALIBABA_SECRET_ACCESS_KEY")
+
+        object_dir = "test_mlrun_oss_objects"
+        object_file = f"file_{random.randint(0, 1000)}.txt"
+        csv_file = f"file_{random.randint(0,1000)}.csv"
+
+        self.oss = {}
+        self.oss["oss"] = self._make_target_names(
+            "oss://", self._bucket_name, object_dir, object_file, csv_file
+        )
```

Comment:
Minor style suggestion:
```suggestion
        self.oss = {
            "oss": self._make_target_names(
                "oss://", self._bucket_name, object_dir, object_file, csv_file
            )
        }
```